File: src/main/java/com/example/kyudo_app/presentation/controller/UserResistorController.java
package com.example.kyudo_app.presentation.controller;

import com.example.kyudo_app.domain.model.User;
import com.example.kyudo_app.domain.service.UserRegisterService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/user")
@CrossOrigin(origins = "http://localhost:5173")
public class UserResistorController {

private final UserRegisterService userRegisterService;

public UserResistorController(UserRegisterService userRegisterService){
    this.userRegisterService = userRegisterService;
}

    @GetMapping("/user")

    public String getUsers() {
        return "ユーザー一覧";
    }
    @PostMapping("/register")
    public ResponseEntity<User> registerUser(@RequestBody User user) {
        User registeredUser = userRegisterService.registryUser(user);
        return ResponseEntity.ok(registeredUser);
    }

    @GetMapping("/{id}")
    public ResponseEntity<User> getUserById(@PathVariable Long id) {
        User user = userRegisterService.getUserById(id);
        return ResponseEntity.ok(user);
    }


}

File: src/main/kotlin/com/example/kyudo_app/presentation/controller/UserResistorController.kt
package com.example.kyudo_app.presentation.controller

import com.example.kyudo_app.domain.model.User
import com.example.kyudo_app.domain.service.UserRegisterService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
@CrossOrigin(origins = ["http://localhost:5173"])
class UserResistorController(private val userRegisterService: UserRegisterService) {
    @get:GetMapping("/user")
    val users: String
        get() = "ユーザー一覧"

    @PostMapping("/register")
    fun registerUser(@RequestBody user: User): ResponseEntity<User?> {
        val registeredUser = userRegisterService.registryUser(user)
        return ResponseEntity.ok<User?>(registeredUser)
    }

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: Long): ResponseEntity<User?> {
        val user = userRegisterService.getUserById(id)
        return ResponseEntity.ok<User?>(user)
    }
}

